### PR TITLE
Better handle null/undefined mode prop

### LINKED ISSUE #452: Can't import modes from @deck.gl/layers
#### Describe the bug
```
import {DrawPolygonMode} from '@nebula.gl/layers'
```

#### Actual Result
Can't call getGuides on undefined

If you change to 
```
import {DrawPolygonMode} from '@nebula.gl/edit-modes'
```
then it works

#### Expected Result
Works

#### Reproduce Steps
https://codesandbox.io/s/deckgl-and-nebulagl-editablegeojsonlayer-no-react-p9yrs

#### To Do List 
  - [x] Add label and assign to milestone
  - [ ] Coding
  - [ ] Test

### PR DESCRIPTION
**Pull request by supersonicclay**
Fixes #452

### CODE REVIEW

**Comment by supersonicclay** [comment]
cc @andybak

File: modules/layers/src/layers/editable-geojson-layer.ts [hunk 1]
```diff
@@ -285,11 +286,17 @@ export default class EditableGeoJsonLayer extends EditableLayer {
...
     super.updateState({ props, changeFlags });
 
-    let mode = this.state.mode;
     if (changeFlags.propsOrDataChanged) {
-      if (props.mode !== oldProps.mode) {
-        if (typeof props.mode === 'string') {
-          // Lookup the mode based on its name (for legacy purposes)
+      const modePropChanged = Object.keys(oldProps).length === 0 || props.mode !== oldProps.mode;
+      if (modePropChanged) {
+        let mode = this.state.mode;
```
Comment (xintongxia) [inline modules/layers/src/layers/editable-geojson-layer.ts:292]: this isn't used, `mode` is set based on `props.mode` later.

File: modules/layers/src/layers/editable-geojson-layer.ts [hunk 2]
```diff
@@ -299,15 +306,10 @@ export default class EditableGeoJsonLayer extends EditableLayer {
           mode = props.mode;
```
Comment (xintongxia) [inline modules/layers/src/layers/editable-geojson-layer.ts:306]: if `mode` isnt `string` or `function`, what it would be?

Comment (supersonicclay) [reply modules/layers/src/layers/editable-geojson-layer.ts:306]: An instance of an EditMode.

**Pull request merged**